addons/product_expiry/tests/test_generate_serial_numbers.py

```diff
@@ -238,3 +239,32 @@ def test_set_multiple_lot_name_with_expiration_date_06_one_line(self):
             self.assertEqual(move.move_line_ids.lot_name, "lot-001")
             self.assertEqual(move.move_line_ids.quantity, 20)
             self.assertEqual(move.move_line_ids.expiration_date, datetime(day=4, month=8, year=2048))
+
+
+class TestProductExpiryTour(TestStockPickingTour):
+
+    def test_generate_serial_with_expiration(self):
+        """
+        Ensure that serial/lot numbers generated using the 'Generate Serials/Lots' button in Detailed
+        Operations have expiration dates set.
+        """
+        product_exp = self.env['product.product'].create({
+            'name': 'Product Exp',
+            'is_storable': True,
+            'tracking': 'serial',
+            'use_expiration_date': True,
+            'expiration_time': 2,
+        })
+
+        self.env['stock.move'].create({
+            'name': product_exp.name,
+            'product_id': product_exp.id,
+            'product_uom_qty': 1,
```

Comment:
Put at least 2  and check all the move lines have the expiration_date set and correct (you can use freeze_time to assert easily on the date value). thank you